src/antares/service/local_services/binding_constraint_local.py

```diff
@@ -47,6 +48,10 @@ def create_binding_constraint(
         equal_term_matrix: Optional[pd.DataFrame] = None,
         greater_term_matrix: Optional[pd.DataFrame] = None,
     ) -> BindingConstraint:
+        if name in self.binding_constraints:
```

Comment:
You should check with the id as it's what you store inside self.binding_constraints